pkg/controller/seed-controller-manager/mla/ratelimit_cortex_controller.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mla
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/kubernetes"
+	"k8c.io/kubermatic/v2/pkg/version/kubermatic"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	runtimeConfigMap      = "cortex-runtime-config"
+	runtimeConfigFileName = "runtime-config.yaml"
+)
+
+type tenantOverride struct {
+	IngestionRate      *int32 `yaml:"ingestion_rate,omitempty"`
+	MaxSeriesPerMetric *int32 `yaml:"max_series_per_metric,omitempty"`
+	MaxSeriesPerQuery  *int32 `yaml:"max_series_per_query,omitempty"`
+	MaxSamplesPerQuery *int32 `yaml:"max_samples_per_query,omitempty"`
+	IngestionBurstSize *int32 `yaml:"ingestion_burst_size,omitempty"`
+	MaxSeriesTotal     *int32 `yaml:"max_series_per_user,omitempty"`
```

Comment:
any reasons of using snake case not camel case :) ?

Reply:
Is it because those keys in cortex are using snake_case?

Reply:
yeah, of course, it's cortex configuration, not our